compaction/compaction_manager.hh

```diff
@@ -119,6 +163,13 @@ public:
         std::string describe() const;
     };
 
+    class major_compaction_task;
+    class custom_compaction_task;
+    class regular_compaction_task;
+    class offstrategy_compaction_task;
+    class rewrite_sstables_compaction_task;
+    class compaction_manager_test_task;
```

Comment:
Any reason these have to be exposed here?

Reply:
Unfortunately they need access to compaction_manager private members.

Reply:
We should cleanup the api the compaction_manager exposes for the tasks, but let's do it as a follow-up.

Reply:
Ok, follow-up is fine.